Automattic/jetpack
# Related Posts: fix accessibility issues

Fixes https://github.com/Automattic/jetpack/issues/34594

## Proposed changes:
This PR fixes several accessibility issues of the _Related Posts_ block in the front-end.

- Add an accessible name to the component `nav` by adding an `aria-label` or `aria-labelled-by` attribute.
- Use `ul` and `li` tags for the markup of items and their container.
- Render all posts in the same list and handle layout in CSS instead.
- Remove the `ul` tag for each list item and use a `dl` list for the metadata only.
- Remove `role="menuitem"` from post items.  It's incorrectly used (see [these docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menuitem_role)).
- Remove the `aria-labelledby` attribute from post items. It's used for interactive elements or landmarks.
- For each post item, merge the two links (image and text) into a single link that contains both the post thumbnail and title.
- Fix thumbnails so they don't appear stretched.
- Adjust spacing to better group post metadata (date, author, and tag).
- Add spacing between items in "list" view mode.

_Note: when set, the headline is rendered as a `h3`. This is wrong since assumptions are made about the outline of the page. Authors should be able to set the heading level themselves in the editor. This should be handled as a separate PR._

### Other information:


## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






### Prerequisites

- Create a couple of posts that share the same tag
- For some of them, add a featured image

### Testing

- Spin up a test site with this branch
- Build the plugin
- Create a new post
- Add the tag you've used for the related posts
- Add the _Related Posts_ block
- Open the preview
- Play with different settings and views in the editor and verify nothing looks broken
- Optionally, test the component with a screen reader

_Settings_
<img width="276" alt="Screenshot 2024-01-09 at 3 20 17 PM" src="https://github.com/Automattic/jetpack/assets/1620183/95a6ce05-7074-422b-9d5b-023c8baf0a91">

_Views_
<img width="283" alt="Screenshot 2024-01-10 at 1 40 01 PM" src="https://github.com/Automattic/jetpack/assets/1620183/8a9cb45b-b67f-46df-b798-ed079300d5b0">

## Captures

<img width="400" alt="Screenshot 2024-01-11 at 11 34 14 AM" src="https://github.com/Automattic/jetpack/assets/1620183/863571ee-6a69-4de1-a471-3794227372a1">
<img width="400" alt="Screenshot 2024-01-11 at 11 35 03 AM" src="https://github.com/Automattic/jetpack/assets/1620183/7d0ca3fe-213b-4bb7-bb60-8e254e50ea54">
